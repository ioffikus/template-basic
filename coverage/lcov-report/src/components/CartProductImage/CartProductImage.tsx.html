<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/CartProductImage/CartProductImage.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/CartProductImage</a> CartProductImage.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/1</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PreloadingImage from 'src/components/PreloadingImage';
&nbsp;
interface IProps {
  pictureUrl: string;
  imageAlt?: string;
}
&nbsp;
interface IState {}
&nbsp;
class CartProductImage extends React.Component&lt;IProps, IState&gt; {
  render() {
    return (
      &lt;div className="root"&gt;
        &lt;style jsx&gt;{`
          .root {
            position: relative;
            display: flex;
            align-items: center;
            justify-content: center;
            height: 50px;
            width: 50px;
            min-width: 50px;
            overflow: hidden;
            border: 1px solid rgba(6, 47, 73, 0.1);
            border-radius: 4px;
            background-color: #f5f8fb;
          }
          .root :global(.image-spinner) {
            width: 100%;
            height: 100%;
          }
          .root :global(.image) {
            position: relative;
            z-index: 2;
            max-height: 100%;
            max-width: 100%;
          }
        `}&lt;/style&gt;
        {this.props.pictureUrl &amp;&amp; (
          &lt;PreloadingImage
            src={this.props.pictureUrl}
            imageClass="image"
            wrapperClass="image-spinner"
            spinSize="small"
            imageAlt={this.props.imageAlt}
          /&gt;
        )}
        {this.props.pictureUrl &amp;&amp; &lt;img className="app-image-blur" src={this.props.pictureUrl} /&gt;}
      &lt;/div&gt;
    );
  }
}
&nbsp;
export default CartProductImage;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Dec 29 2017 11:52:09 GMT+0300 (MSK)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
