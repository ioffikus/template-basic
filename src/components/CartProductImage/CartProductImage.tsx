import React from 'react';
import PreloadingImage from 'src/components/PreloadingImage';

interface IProps {
  pictureUrl: string;
  imageAlt?: string;
}

interface IState {}

class CartProductImage extends React.Component<IProps, IState> {
  render() {
    return (
      <div className="root">
        <style jsx>{`
          .root {
            position: relative;
            display: flex;
            align-items: center;
            justify-content: center;
            height: 50px;
            width: 50px;
            min-width: 50px;
            overflow: hidden;
            border: 1px solid rgba(6, 47, 73, 0.1);
            border-radius: 4px;
            background-color: #f5f8fb;
          }
          .root :global(.image-spinner) {
            width: 100%;
            height: 100%;
          }
          .root :global(.image) {
            position: relative;
            z-index: 2;
            max-height: 100%;
            max-width: 100%;
          }
        `}</style>
        {this.props.pictureUrl && (
          <PreloadingImage
            src={this.props.pictureUrl}
            imageClass="image"
            wrapperClass="image-spinner"
            spinSize="small"
            imageAlt={this.props.imageAlt}
          />
        )}
        {this.props.pictureUrl && <img className="app-image-blur" src={this.props.pictureUrl} />}
      </div>
    );
  }
}

export default CartProductImage;
