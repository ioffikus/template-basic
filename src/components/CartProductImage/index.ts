import CartProductImage from './CartProductImage';

export default CartProductImage;
