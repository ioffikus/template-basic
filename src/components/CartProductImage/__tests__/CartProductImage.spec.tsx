import React from 'react';
import { shallow } from 'enzyme';
import CartProductImage from '../CartProductImage';

describe(null, () => {
  it('render() with image', () => {
    const pictureUrl = 'dummy picture url';
    const imageAlt = 'dummy image name';
    const wrapper = shallow(<CartProductImage pictureUrl={pictureUrl} imageAlt={imageAlt} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('render() without image', () => {
    const pictureUrl = '';
    const wrapper = shallow(<CartProductImage pictureUrl={pictureUrl} />);
    expect(wrapper).toMatchSnapshot();
  });
});
