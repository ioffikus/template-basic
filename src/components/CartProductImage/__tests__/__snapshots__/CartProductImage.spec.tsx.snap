// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render() with image 1`] = `
<div
  className="jsx-1807206759 root"
>
  <JSXStyle
    css=".root.jsx-1807206759{position:relative;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;height:50px;width:50px;min-width:50px;overflow:hidden;border:1px solid rgba(6,47,73,0.1);border-radius:4px;background-color:#f5f8fb;}.root.jsx-1807206759 .image-spinner{width:100%;height:100%;}.root.jsx-1807206759 .image{position:relative;z-index:2;max-height:100%;max-width:100%;}"
    styleId="1807206759"
  />
  <PreloadingImage
    imageAlt="dummy image name"
    imageClass="image"
    spinSize="small"
    src="dummy picture url"
    wrapperClass="image-spinner"
  />
  <img
    className="jsx-1807206759 app-image-blur"
    src="dummy picture url"
  />
</div>
`;

exports[`render() without image 1`] = `
<div
  className="jsx-1807206759 root"
>
  <JSXStyle
    css=".root.jsx-1807206759{position:relative;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;height:50px;width:50px;min-width:50px;overflow:hidden;border:1px solid rgba(6,47,73,0.1);border-radius:4px;background-color:#f5f8fb;}.root.jsx-1807206759 .image-spinner{width:100%;height:100%;}.root.jsx-1807206759 .image{position:relative;z-index:2;max-height:100%;max-width:100%;}"
    styleId="1807206759"
  />
</div>
`;
